starter/class/connect-four.js

```diff
@@ -21,24 +21,88 @@ class ConnectFour {
     Screen.setGridlines(true);
 
     // Replace this with real commands
-    Screen.addCommand('t', 'test command (remove)', ConnectFour.testCommand);
+    // Screen.addCommand('t', 'test command (remove)', ConnectFour.testCommand);
+    Screen.addCommand('up', 'moves cursor up', this.cursor.up);
+    Screen.addCommand('Down', 'moves cursor down', this.cursor.down);
+    Screen.addCommand('left', 'move cursor left', this.cursor.left);
+    Screen.addCommand('right', 'moves cursor right', this.cursor.right);
+    Screen.addCommand('return', "places a move in the current position", this.placeMove)
 
     this.cursor.setBackgroundColor();
     Screen.render();
   }
+  // Return 'X' if player X wins
+  // Return 'O' if player O wins
+  // Return 'T' if the game is a tie
+  // Return false if the game has not ended
 
-  // Remove this
-  static testCommand() {
-    console.log("TEST COMMAND");
+  placeMove = () => {
+    if (Screen.grid[this.cursor.row][this.cursor.col] === " ") {
+      Screen.setTextColor(this.cursor.row, this.cursor.col, 'white');
+      Screen.setGrid(this.row, this.col, this.playerTurn);
+      Screen.render()
+      if (Cursor.checkWin(Screen.grid)) {
+        Cursor.endGame(Cursor.checkWin(Screen.grid));
+      }
```

Comment:
after you solve the above, you'll want to ask yourself if this is correct.

Reply:
I understand what I was doing wrong here as well.